src/emit/binary.rs

```diff
@@ -1055,7 +1066,7 @@ impl<'a> Binary<'a> {
     ) -> PointerValue<'a> {
         match array_ty {
             Type::Array(_, dim) => {
-                if dim[0].is_some() {
+                if matches!(dim[0], ArrayLength::Fixed(_)) {
```

Comment:
Shouldn't you check the last dimension here?

Reply:
Handled in your pr